karapace/backup/api.py

```diff
@@ -384,16 +392,30 @@ def _handle_producer_send(
         instruction.key,
         instruction.value,
     )
-    producer.send(
-        instruction.topic_name,
-        key=instruction.key,
-        value=instruction.value,
-        partition=instruction.partition_index,
-        headers=[(key.decode() if key is not None else None, value) for key, value in instruction.headers],
-        timestamp_ms=instruction.timestamp,
-    ).add_errback(_raise_backup_error)
+    try:
+        producer.send(
+            instruction.topic_name,
+            key=instruction.key,
+            value=instruction.value,
+            partition=instruction.partition_index,
+            headers=[(key.decode() if key is not None else None, value) for key, value in instruction.headers],
+            timestamp_ms=instruction.timestamp,
+        ).add_errback(_raise_backup_error)
+    except (KafkaError, AssertionError) as ex:
+        raise BackupDataRestorationError("Error while calling send on restoring messages") from ex
+
+
+@contextlib.contextmanager
+def _enable_producer_callback_errors() -> Iterator[Callable[..., None]]:
```

Comment:
```suggestion
def _enable_producer_callback_errors() -> Iterator[None]:
```

We're not yielding anything from this function, just `None` :)